### PR TITLE
Add custom CSE emoticons

### PR DESCRIPTION
**Pull request by JoelTerMeer**
Added CSE emoticons, changed regex, created case-sensitive and case-insensitive sections for commands, and added suggested abbreviations and variants to commands.

### CODE REVIEW

File: patcher/src/ts/content/chat/ParseEmoji.tsx [hunk 1]
```diff
@@ -7,22 +7,23 @@
 import * as React from 'react';
 
 function emojiNameFromText(text:string): string {
-  switch(text) {
-    case ':angry:': case ':[': case ':-[':
```
Comment (JoelTerMeer) [inline patcher/src/ts/content/chat/ParseEmoji.tsx]: :[ and :-[ didn't work before regex change

File: patcher/src/ts/content/chat/ParseEmoji.tsx [hunk 2]
```diff
@@ -59,7 +139,7 @@ function fromText(text: string, keygen: () => number) : JSX.Element[] {
 }
 
 function createRegExp() : RegExp {
-  return /[3]*[;:8][-']*[()@oO#$*pPD/|><]|\([6aAhH]\)|-_-zzZ|\(-_-\)zzZ|[oO]\.[oO]|:[a-z]*:/g;
+  return /:[a-zA-Z0-9]+:|[3]*[;:8][-']*[()@oO#$*pPD/|><]|\([6aAhH]\)|-_-zzZ|\(-_-\)zzZ|[oO]\.[oO]/g;
```
Comment (JoelTerMeer) [inline patcher/src/ts/content/chat/ParseEmoji.tsx]: changed regex to mehuge/agoknee recommendation to allow :pvp: to not return :P + "vp:"

File: patcher/src/ts/content/chat/ParseEmoji.tsx [hunk 3]
```diff
@@ -35,18 +36,97 @@ function emojiNameFromText(text:string): string {
     case ':shocked:': case '8o': case '8-o': case '8-O':
```
Comment (sysrage) [inline patcher/src/ts/content/chat/ParseEmoji.tsx]: Remove '8-O'

File: patcher/src/ts/content/chat/ParseEmoji.tsx [hunk 4]
```diff
@@ -35,18 +36,97 @@ function emojiNameFromText(text:string): string {
...
     case ':O': case ':o':
```
Comment (sysrage) [inline patcher/src/ts/content/chat/ParseEmoji.tsx]: Remove ':O'

File: patcher/src/ts/content/chat/ParseEmoji.tsx [hunk 5]
```diff
@@ -35,18 +36,97 @@ function emojiNameFromText(text:string): string {
...
       return 'shocked';
-    case ':sleepy:': case '-_-zzZ': case '(-_-)zzZ':
+    case ':sleepy:': case '-_-zzZ': case '(-_-)zzZ': case ':zzz:':
```
Comment (sysrage) [inline patcher/src/ts/content/chat/ParseEmoji.tsx]: Change the 'zzZ' to 'zzz' or move to case sensitive section.

**Comment by sysrage** [comment]
Looking good. Only a few minor suggestions. :)

**Pull request merged**